### PR TITLE
modify TypeError

### PR DESCRIPTION
**Pull request by jjaegii**
[ TypeError: only integer scalar arrays can be converted to a scalar index ] at 'selected_poses.append(poses[index])' in 'select_2_poses'

### CODE REVIEW

File: projects/perception/skeleton_based_action_recognition/demos/demo.py [hunk 1]
```diff
@@ -92,6 +91,7 @@ def pose2numpy(args, num_current_frames, poses_list):
     return skeleton_seq
 
 
+# TypeError: only integer scalar arrays can be converted to a scalar index (complete)
```
Comment (stefaniapedrazzi) [inline projects/perception/skeleton_based_action_recognition/demos/demo.py:94]: This comment seems more a development message and should probably be removed.

File: projects/perception/skeleton_based_action_recognition/demos/demo.py [hunk 2]
```diff
@@ -100,7 +100,8 @@ def select_2_poses(poses):
         energy.append(s)
     energy = np.array(energy)
     index = energy.argsort()[::-1][0:2]
-    selected_poses.append(poses[index])
+    # selected_poses.append(poses[index])
```
Comment (stefaniapedrazzi) [inline projects/perception/skeleton_based_action_recognition/demos/demo.py:103]: This line should be removed

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
Thank you. The fix is fine.
But there are some additional changes that doesn't seem needed.

File: projects/perception/skeleton_based_action_recognition/demos/demo.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ class VideoReader(object):
     def __init__(self, file_name):
         self.file_name = file_name
         try:  # OpenCV needs int to read from webcam
-            self.file_name = int(file_name)
+            self.file_name = file_name
```
Comment (passalis) [inline projects/perception/skeleton_based_action_recognition/demos/demo.py:38]: This try block will always succeed after the change. So, if we do not need this it should be altogether removed. Why is this change needed?

Comment (jjaegii) [reply projects/perception/skeleton_based_action_recognition/demos/demo.py:38]: because you can bring not only webcams but also other videos. for example, rtsp(cctv) streams.

File: projects/perception/skeleton_based_action_recognition/demos/demo.py (hunk 2 above)
Comment (jjaegii) [reply projects/perception/skeleton_based_action_recognition/demos/demo.py:103]: i will fix it soon, thank you

**Comment by jjaegii** [comment]
change cuda to cpu,
remove useless comment.

File: projects/perception/skeleton_based_action_recognition/demos/demo.py (hunk 3 above)
Comment (passalis) [reply projects/perception/skeleton_based_action_recognition/demos/demo.py:38]: In the modified version of the code the `try` block can be removed altogether, since the assignment has already been done in l.36. Removing lines 37-40 in the current version (without the int cast) will not change anything.

However, I was trying to understand why this block exists in the first place.   To the best of my understanding, the behavior of VideoCapture changes according to the data types it receives (https://docs.opencv.org/3.4/d8/dfe/classcv_1_1VideoCapture.html). When an `int` is passed, then it tries to open the corresponding camera. On the other hand, when a string is passed, then it tries to open the corresponding stream or file. The `try`... `except` block is essentially trying to cope this behavior - in a not very clear way (especially given that the argument is called `file_name`). If the supplied `file_name` string can be cast into an `int`, then it implicitly forces opencv to read from the corresponding webcam. Of course, if there is a file with no suffix and can be cast into a number, then this code will not read from the file and it will try to open the corresponding webcam. However, in any other case it should function correctly. For example, you shouldn't have issues with reading from rtsp streams using the existing version of the code.

Comment (jjaegii) [reply projects/perception/skeleton_based_action_recognition/demos/demo.py:38]: oh, i see!
